client/specs/booking.test.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import sinon from 'sinon';
+import Booking from '../src/components/Booking.jsx';
+import Form from '../src/components/Form.jsx';
+import Price from '../src/components/Price.jsx';
+
+describe('<Booking />', () => {
+  test('Should fetch all room data from the server upon mounting', () => {
+    sinon.spy(Booking.prototype, 'getRoomData');
+    const wrapper = mount( < Booking room={5} /> );
+    expect(Booking.prototype.getRoomData.calledOnce).toBe(true);
+    Booking.prototype.getRoomData.restore();
+  });
+});
+
+describe('<Form />', () => {
+  test('Should send user info to the server when user clicks the Book button', () => {
+    const onButtonClick = sinon.spy();
+    sinon.spy(Form.prototype, 'sendBookingRequest');
+    const wrapper = mount((
+      <Form onButtonClick={onButtonClick} />
+    ));
+    wrapper.find('button').simulate('click');
+    expect(Form.prototype.sendBookingRequest.calledOnce).toBe(true);
+    Form.prototype.sendBookingRequest.restore();
+  });
+
+  test('Should get/set props for other components', () => {
```

Comment:
Last test seems unnecessary, if you pass in a certain a prop, it will definitely be the value that you passed in. You should test your component to see what it does in response to receiving a prop, or receiving a prop change.

Reply:
@leealbert95 Good point. I'll modify this test.